src/components/button-toggle/button-toggle.spec.ts

```diff
@@ -0,0 +1,346 @@
+} from '@angular/core/testing';
+import {TestComponentBuilder, ComponentFixture} from '@angular/compiler/testing';
+import {Component, DebugElement, provide} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {
+  MD_BUTTON_TOGGLE_DIRECTIVES,
+  MdButtonToggleGroup,
+  MdButtonToggle,
+  MdButtonToggleGroupMultiple
+} from './button-toggle';
+import {
+  MdUniqueSelectionDispatcher
+} from '@angular2-material/core/coordination/unique-selection-dispatcher';
+
+
+describe('MdButtonToggle', () => {
+  let builder: TestComponentBuilder;
+  let dispatcher: MdUniqueSelectionDispatcher;
+
+  beforeEachProviders(() => [
+    provide(MdUniqueSelectionDispatcher, {useFactory: () => {
+      dispatcher = new MdUniqueSelectionDispatcher();
+      return dispatcher;
+    }})
+  ]);
+
+  beforeEach(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
+    builder = tcb;
+  }));
+
+  describe('inside of an exclusive selection group', () => {
+    let fixture: ComponentFixture<ButtonTogglesInsideButtonToggleGroup>;
+    let groupDebugElement: DebugElement;
+    let groupNativeElement: HTMLElement;
+    let buttonToggleDebugElements: DebugElement[];
+    let buttonToggleNativeElements: HTMLElement[];
+    let groupInstance: MdButtonToggleGroup;
+    let buttonToggleInstances: MdButtonToggle[];
+    let testComponent: ButtonTogglesInsideButtonToggleGroup;
+
+    beforeEach(async(() => {
+      builder.createAsync(ButtonTogglesInsideButtonToggleGroup).then(f => {
+        fixture = f;
+        fixture.detectChanges();
+
+        testComponent = fixture.debugElement.componentInstance;
+
+        groupDebugElement = fixture.debugElement.query(By.directive(MdButtonToggleGroup));
+        groupNativeElement = groupDebugElement.nativeElement;
+        groupInstance = groupDebugElement.injector.get(MdButtonToggleGroup);
+
+        buttonToggleDebugElements = fixture.debugElement.queryAll(By.directive(MdButtonToggle));
+        buttonToggleNativeElements =
+            buttonToggleDebugElements.map(debugEl => debugEl.nativeElement);
+        buttonToggleInstances = buttonToggleDebugElements.map(debugEl => debugEl.componentInstance);
+      });
+    }));
+
+    it('should set individual button toggle names based on the group name', () => {
+      expect(groupInstance.name).toBeTruthy();
+      for (let buttonToggle of buttonToggleInstances) {
+        expect(buttonToggle.name).toBe(groupInstance.name);
+      }
+    });
+
+    it('should disable click interactions when the group is disabled', () => {
+      testComponent.isGroupDisabled = true;
+      fixture.detectChanges();
+
+      buttonToggleNativeElements[0].click();
+      expect(buttonToggleInstances[0].checked).toBe(false);
+    });
+
+    it('should update the group value when one of the toggles changes', () => {
+      expect(groupInstance.value).toBeFalsy();
+      let nativeCheckboxLabel =
+          <HTMLLabelElement> buttonToggleNativeElements[0].querySelector('label');
```

Comment:
Is it possible for you to use the `fixture.debugElement.query(By.css('label))` for this instead?  If we ever run the tests in a different context, it wouldn't work.